### PR TITLE
feat: edit project slug

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-web#2115

Add:
- `Settings` edit page and `SettingsForm`
- `DebouncedField`
- Update the finished step in the creation flow

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. try to edit a project slug https://deploy-preview-2164--regen-marketplace.netlify.app/project-pages/C01-001/edit/settings
2. test the shortcut from the finished step in the project creation https://deploy-preview-2164--regen-marketplace.netlify.app/project-pages/be755d18-612d-11ee-9603-0267c2be097b/choose-credit-class

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie you can test the slug editing here: https://deploy-preview-2164--regen-marketplace.netlify.app/project-pages/C01-001/edit/settings

**Comment by erikalogie** [comment]
> @erikalogie you can test the slug editing here: https://deploy-preview-2164--regen-marketplace.netlify.app/project-pages/C01-001/edit/settings

I'm having this issue: https://monosnap.com/file/bDcxKaXmHP9fZPzjNu0gVxDSBvr3kD

File: netlify.toml [hunk 1]
```diff
@@ -45,6 +47,7 @@
   VITE_SANITY_DATASET = "staging"
   VITE_RECAPTCHA_SITE_KEY = "6LchxlAaAAAAANjtO3w4b5ihVCL87tYfbe0ltzp9"
   VITE_API_URI = "https://api-staging.regen.network"
+  VITE_APP_URL="http://localhost:3000"
```
Comment (blushi) [inline netlify.toml:50]: why not just use `window.location.origin`?

Comment (flagrede) [reply netlify.toml:50]: Fair point, I think it should be better indeed.

**Comment by flagrede** [comment]
@erikalogie that`s so weird I don't have the same behavior.

@blushi are you able to edit the slug correctly here? https://deploy-preview-2164--regen-marketplace.netlify.app/project-pages/C01-001/edit/settings

File: netlify.toml (hunk 1 above)
Comment (blushi) [reply netlify.toml]: yes we already use this in other places in the app so best to have it consistent and we can migrate everything to be ssr friendly when needed

**Comment by blushi** [comment]
> @erikalogie that`s so weird I don't have the same behavior.
> 
> @blushi are you able to edit the slug correctly here? https://deploy-preview-2164--regen-marketplace.netlify.app/project-pages/C01-001/edit/settings

Yes I could reproduce this with C01-005 project: https://deploy-preview-2164--regen-marketplace.netlify.app/project-pages/C01-005/edit/settings

**Comment by erikalogie** [comment]
> @blushi @erikalogie Ok, there was an issue with on-chain project without an off-chain project. Not sure if it is supposed to happen but it is fixed.

This looks great!

File: web-marketplace/.env.example [hunk 2]
```diff
@@ -1,6 +1,7 @@
 VITE_MAPBOX_TOKEN=
 VITE_STRIPE_API_KEY=
 VITE_WEBSITE_URL=https://www.regen.network
+VITE_APP_URL=https://localhost:3000
```
Comment (blushi) [inline web-marketplace/.env.example:4]:
```suggestion
```

**Comment by flagrede** [comment]
@blushi Make sense, I fixed that.

File: web-marketplace/src/pages/ProjectFinished/ProjectFinished.tsx [hunk 3]
```diff
@@ -61,11 +66,40 @@ const ProjectFinished: React.FC<React.PropsWithChildren<unknown>> = () => {
...
           />
+          <CardItem
+            label="url"
+            value={{
+              name: projectUrl,
+              url: projectUrl,
+              children: (
+                <Box>
+                  <Link
+                    href={`${window.location.origin}/project-pages/${currentProjectId}/edit/settings`}
```
Comment (blushi) [inline web-marketplace/src/pages/ProjectFinished/ProjectFinished.tsx:77]: not sure if we need the window.location.origin here
from the project create flow "finished" step, this opens up the settings page in a new tab, is this what we want @erikalogie?

Comment (flagrede) [reply web-marketplace/src/pages/ProjectFinished/ProjectFinished.tsx:77]: I don't think so indeed.

**Review by blushi — APPROVED** [review]
tACK

**Pull request merged**